network.go

```diff
@@ -504,3 +646,161 @@ func (n *network) getController() *controller {
 	defer n.Unlock()
 	return n.ctrlr
 }
+
+func (n *network) ipamAllocate() ([]func(), error) {
```

Comment:
@tomdee 
Given this PR is big enough and libnetwork <-> ipam driver interaction is on only for overlay driver, I'd address the v6 part as part of a subsequent PR.

For your reference, consider libnetwork will do for v6 what it does for v4, besides auto allocating a pool if it was not explicitely requested via `network.ipamV6Config`